tensorstore/internal/http/http_request.h

```diff
@@ -62,11 +70,16 @@ class HttpRequest {
 /// Implements the builder pattern for HttpRequest.
 class HttpRequestBuilder {
  public:
-  /// Creates a request builder, using the specified method and url.
+  using UriEncodeFunctor = absl::FunctionRef<std::string(std::string_view)>;
+  /// Creates a request builder, using the specified method, url and URI encoding function.
   ///
   /// The method should be an HTTP method, like "GET", "POST", "PUT", "HEAD",
   /// etc.
-  explicit HttpRequestBuilder(std::string_view method, std::string base_url);
+  explicit HttpRequestBuilder(std::string_view method, std::string base_url)
+    : HttpRequestBuilder(method, base_url,
+        [](std::string_view d) { return internal::PercentEncodeUriComponent(d); }) {}
```

Comment:
Used a lambda because the `absl::FunctionRef` constructor struggled to resolve to multiple overloads of `PercentEncodeUriComponent`.